### PR TITLE
COMP: guard '_LARGEFILE_SOURCE' to reduce warnings

### PR DESCRIPTION
**Pull request by ihnorton**
This definition should be guarded. Some other builds/projects/etc. already define `_LARGEFILE_SOURCE`, thus inclusions of `pyconfig.h` will generate a number of warnings [1,2].

(I know this is ultimately from upstream, but we control it here and can at least reduce the warning situation for projects in CMake world)

[1]

```
In file included from Modules/Loadable/InteractiveSeeding/qSlicerTractographyInteractiveSeedingModuleWidgetGenericTest.cxx:47:
In file included from /Users/inorton/git/slcr/s4/Base/QTGUI/qSlicerPythonManager.h:17:
In file included from /Users/inorton/git/slcr/s4/Base/QTCore/qSlicerCorePythonManager.h:25:
In file included from /Users/inorton/git/slcr/s4nj/CTK/Libs/Scripting/Python/Core/ctkAbstractPythonManager.h:31:
In file included from /Users/inorton/git/slcr/s4nj/CTK-build/CMakeExternals/Install/include/PythonQt/PythonQtPythonInclude.h:90:
In file included from /Users/inorton/git/slcr/s4nj/python-install/include/python2.7/Python.h:8:
/Users/inorton/git/slcr/s4nj/python-install/include/python2.7/pyconfig.h:1163:9: warning: '_LARGEFILE_SOURCE' macro redefined [-Wmacro-redefined]
#define _LARGEFILE_SOURCE 1
        ^
/Users/inorton/git/slcr/s4nj/VTKv7/Common/Core/vtkWin32Header.h:36:13: note: previous definition is here
#    define _LARGEFILE_SOURCE
```

[2] tens of warnings in Slicer source build: http://slicer.cdash.org/viewBuildError.php?type=1&buildid=911731

### CODE REVIEW

**Comment by ihnorton** [comment]
cc @jcfr

**Comment by jcfr** [comment]
Thanks for the reminder, I will get to that shortly.

**Comment by jcfr** [comment]
@ihnorton I added a follow up commit. Let me know what you think.

Thanks
Jc

File: cmake/config-unix/pyconfig.h.in [hunk 1]
```diff
@@ -1160,7 +1160,13 @@
 #cmakedefine _GNU_SOURCE 1
 
 /* This must be defined on some systems to enable large file support. */
-#cmakedefine _LARGEFILE_SOURCE @_LARGEFILE_SOURCE@
+#cmakedefine _PY_LARGEFILE_SOURCE @_LARGEFILE_SOURCE@
+#if _LARGEFILE_SOURCE != _PY_LARGEFILE_SOURCE
+# error "_LARGEFILE_SOURCE is inconsistently defined"
```
Comment (ihnorton) [inline cmake/config-unix/pyconfig.h.in]: is it necessary to make this an error? the check should never be anything other than `ifdef`/`ifndef`.

Comment (jcfr) [inline cmake/config-unix/pyconfig.h.in]: Moving forward I would like most of the definition in pyconfig to due a consistency check. 

Look like I missed something here, I basically need to create a macro and test it. 

I will keep this for later and integrate your patch as is.

**Comment by jcfr** [comment]
Integrated in https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/commit/39ccab04fc48d957f93d83f056e65a1dfb09d784

**Pull request closed without merging**